src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,217 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+
+    class Node<E> {
```

Comment:
put inner class at the very end of the enclosing class
make it private and static